flows/src/main/java/com/softwaremill/jox/flows/Flow.java

```diff
@@ -0,0 +1,322 @@
+package com.softwaremill.jox.flows;
+
+
+import com.softwaremill.jox.Channel;
+import com.softwaremill.jox.Source;
+import com.softwaremill.jox.structured.UnsupervisedScope;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static com.softwaremill.jox.flows.Flows.usingEmit;
+import static com.softwaremill.jox.structured.Scopes.unsupervised;
+
+/**
+ * Describes an asynchronous transformation pipeline. When run, emits elements of type `T`.
+ * <p>
+ * A flow is lazy - evaluation happens only when it's run.
+ * <p>
+ * Flows can be created using the {@link Flows#fromValues}, {@link Flows#fromSource}} and other `Flow.from*` methods, {@link Flows#tick} etc.
+ * <p>
+ * Transformation stages can be added using the available combinators, such as {@link Flow#map}, {@link Flow#buffer}, {@link Flow#grouped}, etc.
+ * Each such method returns a new immutable {@link Flow} instance.
+ * <p>
+ * Running a flow is possible using one of the `run*` methods, such as {@link Flow#runToList}, {@link Flow#runToChannel} or {@link Flow#runFold}.
+ */
+public class Flow<T> {
+    protected final FlowStage<T> last;
+
+    public Flow(FlowStage<T> last) {
+        this.last = last;
+    }
+
+    // region Run operations
+
+    /** Invokes the given function for each emitted element. Blocks until the flow completes. */
+    public void runForeach(Consumer<T> sink) throws Exception {
+        last.run(sink::accept);
+    }
+
+    /** Invokes the provided {@link FlowEmit} for each emitted element. Blocks until the flow completes. */
+    public void runToEmit(FlowEmit<T> emit) throws Exception {
+        last.run(emit);
+    }
+
+    /** Accumulates all elements emitted by this flow into a list. Blocks until the flow completes. */
+    public List<T> runToList() throws Exception {
+        List<T> result = new ArrayList<>();
+        runForeach(result::add);
+        return result;
+    }
+
+    /** The flow is run in the background, and each emitted element is sent to a newly created channel, which is then returned as the result
+     * of this method.
+     * <p>
+     * By default, buffer capacity is unlimited.
+     * <p>
+     * Blocks until the flow completes.
```

Comment:
this comment is incorrect/lacking:
1. the channel is rendezvous (capacity 0), not unlimited. Rendezvous channels might mean poor capacity, so I think we should instead by default use a 16-buffered channel (as in Ox). Question is, how to provide this data. Explicit parameters are ... explicit, and not that nice. Scope-local value, maybe? This could be used in other places a swell
2. the method doesn't block - that's also incorrect in Ox's docs, I'll fix it in a second
3. there's no mention on why you need a supervision scope - and I think this might not be obvious to users

Reply:
1. In the next MR, I've added overloaded method with buffer capacity as additional parametr, but there are more methods with similar problems.
I've added ScopedValue in Channel together with default size. 
I'll also add overloaded method in second MR to keep flexibility. 

We might want to consider adding Value Object for buffer capacity to keep things like unlimited buffer, rendevous size, default size, run with scoped etc.

2. Fixed
3. Added, please review if is understandable